python/ray/train/_internal/worker_group.py

```diff
@@ -343,9 +343,21 @@ def add_workers(self, num_workers: int):
         """
         new_actors = []
         new_actor_metadata = []
-        for _ in range(num_workers):
+        for i in range(num_workers):
+            actor_options = {
+                "placement_group": self._placement_group,
+            }
+            if (
+                isinstance(self._placement_group, PlacementGroup)
+                and self._placement_group.bundle_count == num_workers
+            ):
+                actor_options.update(
+                    {
+                        "placement_group_bundle_index": i,
```

Comment:
This will technically affect the existing flow as well, but I _believe_ the logic should still be okay. Let's keep an eye out on tests.

Reply:
Ok, tests have all passed.